scripts/dts/python-devicetree/src/devicetree/edtlib.py

```diff
@@ -2591,9 +2620,8 @@ def _check_prop_by_type(prop_name: str,
     default = options.get("default")
     const = options.get("const")
 
-    if prop_type is None:
-        _err(f"missing 'type:' for '{prop_name}' in 'properties' in "
-             f"{binding_path}")
+    if not prop_type:
+        return
```

Comment:
This seems a little fishy. We shouldn't ignore the checks below when the type hasn't been set yet.

Do you think we could instead move the `self._check()` in `Bindings.__init__()` further down, once all the types have been assigned?

Reply:
let me check, agreed this is not ideal and can skip real problems that were detected before.

Reply:
still looking into this. to make things easier, what we should do is to do the pattern property merges (now done after lookup tables) early in the process.

Reply:
should be fixed in latest revision, let's see if I haven't created more issues :-)